### DISCUSSION TITLE
✨ An option to have a more verbose help output for a function

### DISCUSSION THREAD

**Discussion by ghost** [dagger/dagger#12301] [Ideas]
### What are you trying to do?

This request was inspired by [the addition of Enum support in 0.12](https://dagger.io/blog/dagger-0-12).

```python
@dagger.enum_type
class Severity(dagger.Enum):
    """Vulnerability severity levels"""

    UNKNOWN = "UNKNOWN", "Undetermined risk; analyze further"
    LOW = "LOW", "Minimal risk; routine fix"
    MEDIUM = "MEDIUM", "Moderate risk; timely fix"
    HIGH = "HIGH", "Serious risk; quick fix needed."
    CRITICAL = "CRITICAL", "Severe risk; immediate action."

@dagger.object_type
class Trivy:
    @dagger.function
    async def scan(self, severity: Severity) -> str:
```

Currently, calling help on a function that accepts an Enum as an argument just returns a list of all the enum variants. For example, calling `dagger call scan --help` would return something along the lines of:

```
...
ARGUMENTS
                --severity UNKNOWN,LOW,MEDIUM,HIGH,CRITICAL
```

It would be great if we could get a more verbose output which also prints the description of each enum variant, with one enum variant per line for better readability:

```
ARGUMENTS
                --severity
                     - UNKNOWN                Undetermined risk; analyze further
                     - LOW                           Minimal risk; routine fix
                     - MEDIUM                     Moderate risk; timely fix
                     - HIGH                          Serious risk; quick fix needed.
                     - CRITICAL                   Severe risk; immediate action.
```

Some other CLI tools have this notion of help verbosity.
Perhaps, it could be done through a flag?
`dagger call scan --help -v` or `dagger call scan --vhelp`
Not sure of the best way to add it to the current CLI API, but this option would be helpful.

<hr>

@helderco gave an example of how it's handled within the `uv` tool.

`uv help run` returns verbose help output:
```
--color <COLOR_CHOICE>
          Control colors in output
          
          [default: auto]

          Possible values:
          - auto:   Enables colored output only when the output is going to a terminal or TTY with
            support
          - always: Enables colored output regardless of the detected environment
          - never:  Disables colored output
```
`uv run --help` will show a much more concise help section for the `color` option:
```
--color <COLOR_CHOICE>
          Control colors in output [default: auto] [possible values: auto, always, never]
```

### Why is this important to you?

Ideally, someone who wants to run or play around with pipelines (but didn't author them) in a project would get all the info they need, or at least as much as possible, through the Dagger CLI. 

It enhances the pipeline documentation and stops people from having to read the source code.

**Comment by helderco** [comment]
Some CLIs also make a distinction between `-h` (concise) and `--help` (more info).
